tutorial/cache_pyh_files.md

```diff
@@ -0,0 +1,32 @@
+# Picklize Header files
+
+## Header files
+
+A header file in Pyccel is a file containing function/variable declarations, macro definitions , templates and metavariable declarations.\
+the rule is to give header files names that end with `.pyh` .\
+Header files serve two purposes:
+-  Link external libraries in the targeted languages by providing their function definitions.
+-  Accelerate the parsing process by parsing the header file instead of the original file in the case of pyccelizing multiple files.
+
+### Example
+We create the file `header.pyh` that contains an openmp function definiton:
+
+```python
+#$ header metavar module_name = 'omp_lib'
+#$ header metavar import_all  = True
+
+#$ header function omp_get_num_threads() results(int)
+```
+We then create `openmp.py` file:
+
+```python
+from header import omp_get_num_threads
+print('number of threads is :', omp_get_num_threads())
+```
+Pyccel can compile the Python file with the following command: `pyccel openmp.py --language fortran --flags="-fopenmp"`
```

Comment:
Don't we pass the `-fopenmp` flag automatically?

Reply:
this is a general example pyccel doesn't know that it's an openmp

Reply:
But if the user writes
```bash
$ pyccel --help
```
he/she will discover that we have an `--openmp` flag available.

Reply:
Not in pyccel. Maybe with epyccel, but we are passing `accelerator='openmp'` everytime in `tests_epyccel_openmp.py` so I'm guessing not

Reply:
yes, don't focus too much on the example. I could have used another library or created my own fortran function and then link it using the same mechanism